File: domain/src/main/java/domain/Vehicle.java
package domain;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;

@Entity
public class Vehicle implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String hardwareSerialNumber;
    @Enumerated(EnumType.STRING)
    private VehicleType vehicleType;
    @ManyToOne
    private List<Activity> activities;
    @OneToMany
    private Civilian owner;
    @OneToMany
    private Location currentLocation;

    public Vehicle() {}

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getHardwareSerialNumber() {
        return hardwareSerialNumber;
    }

    public void setHardwareSerialNumber(String hardwareSerialNumber) {
        this.hardwareSerialNumber = hardwareSerialNumber;
    }

    public VehicleType getVehicleType() {
        return vehicleType;
    }

    public void setVehicleType(VehicleType vehicleType) {
        this.vehicleType = vehicleType;
    }

    public List<Activity> getActivities() {
        return activities;
    }

    public void setActivities(List<Activity> activities) {
        this.activities = activities;
    }

    public Civilian getOwner() {
        return owner;
    }

    public void setOwner(Civilian owner) {
        this.owner = owner;
    }
}

File: domain/src/main/java/domain/Vehicle.kt
package domain

import domain.enums.VehicleType

import javax.persistence.*
import java.io.Serializable
import java.util.ArrayList

@Entity
class Vehicle : Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id: Long? = null

    var hardwareSerialNumber: String? = null
    var licensePlate: String? = null
    @Enumerated(EnumType.STRING)
    var vehicleType: VehicleType? = null
    @OneToMany(cascade = CascadeType.ALL)
    var activities: List<Activity>? = null
    @ManyToOne(cascade = CascadeType.ALL)
    var owner: Profile? = null
    @ManyToOne(cascade = CascadeType.ALL)
    var currentLocation: Location? = null

    constructor() {}

    constructor(hardwareSerialNumber: String, vehicleType: VehicleType, currentLocation: Location) {
        this.hardwareSerialNumber = hardwareSerialNumber
        this.vehicleType = vehicleType
        this.currentLocation = currentLocation
        this.activities = ArrayList()
    }

    companion object {

        private val serialVersionUID = 1L
    }
}
